src/transform/mod.rs

```diff
@@ -71,6 +71,29 @@ pub enum TxType {
   H_FLIPADST = 15,
 }
 
+impl TxType {
+  /// Compute transform type for inter chroma.
+  ///
+  /// <https://aomediacodec.github.io/av1-spec/#compute-transform-type-function>
+  #[inline]
+  pub fn uv_inter(self: Self, uv_tx_size: TxSize) -> Self {
+    use TxType::*;
+    if uv_tx_size.sqr_up() == TX_32X32 {
+      match self {
+        IDTX => IDTX,
+        _ => DCT_DCT,
+      }
+    } else if uv_tx_size.sqr() == TX_16X16 {
+      match self {
+        V_ADST | H_ADST | V_FLIPADST | H_FLIPADST => DCT_DCT,
+        _ => self,
+      }
+    } else {
+      self
```

Comment:
I mean, uv_tx_size is 64x64 (when input is chroma subsampling 444) falls at here, which is wrong, correct? Becase only dct_dct is allowed for 64x64.

Reply:
Because only DCT is allowed for 64x64, both for Y and UV, `self` is correct.

Reply:
That is right, but the tx_loop in rdo_tx_type... ( ) only iterate for dct_dct? If so, this is fine.

Reply:
Implicitly, yes. We filter the candidate `TxType` values by membership of the `TxSet`.
https://github.com/xiph/rav1e/blob/p20200825/src/rdo.rs#L1682-L1686
Conversely, if we didn't do this then we would be testing invalid transform types for the luma plane.

Reply:
Got it. I double checked that, inside pub fn rdo_tx_type_decision( ),
it checks
    if av1_tx_used[tx_set as usize][tx_type as usize] == 0 {
      continue;
    }

so, the tx_type loop does not test other than dct_dct for 64x64.